### PR TITLE
Security Scans: Improving telemetry error messages 

### PR DESCRIPTION
**Pull request by laileni-aws**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
- Security Scans: Improving telemetry error messages 

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt [hunk 1]
```diff
@@ -323,14 +323,26 @@ class CodeWhispererCodeScanManager(val project: Project) {
         return errorMessage
     }
 
+    private fun handleCodeScanException(e: CodeWhispererCodeScanException): String? {
```
Comment (ctlai95) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt:326]: nit: this is not handling anything, a better name might be `getCodeScanExceptionMessage`

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt [hunk 2]
```diff
@@ -323,14 +323,26 @@ class CodeWhispererCodeScanManager(val project: Project) {
...
+        return when {
+            message == message("codewhisperer.codescan.invalid_source_zip_telemetry") -> {
+                message("codewhisperer.codescan.run_scan_error")
+            }
+            message.isNullOrBlank() -> null
+            else -> message
+        }
+    }
+
+    private fun handleCodeScanServerException(e: CodeWhispererCodeScanServerException): String? =
```
Comment (ctlai95) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt:337]: same here

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt [hunk 3]
```diff
@@ -372,10 +384,14 @@ class CodeWhispererCodeScanManager(val project: Project) {
...
             }
-            is UploadCodeScanException -> e.message
+            is CodeWhispererCodeScanServerException -> e.message
             is WaiterTimeoutException, is TimeoutCancellationException -> message("codewhisperer.codescan.scan_timed_out")
             is CancellationException -> message("codewhisperer.codescan.cancelled_by_user_exception")
-            else -> e.message
+            else -> when {
+                e.message?.startsWith("codewhisperer.codescan.run_scan_throttling_error_telemetry") == true -> message("codewhisperer.codescan.run_scan_throttling_error_telemetry")
+                e.message != null -> e.message
+                else -> null
```
Comment (ctlai95) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt:393]: isn't this the same as
```
else -> e.message
```

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt [hunk 4]
```diff
@@ -372,10 +384,14 @@ class CodeWhispererCodeScanManager(val project: Project) {
                 message("codewhisperer.codescan.file_too_large") -> message("codewhisperer.codescan.file_too_large_telemetry")
                 else -> e.message
             }
-            is UploadCodeScanException -> e.message
+            is CodeWhispererCodeScanServerException -> e.message
             is WaiterTimeoutException, is TimeoutCancellationException -> message("codewhisperer.codescan.scan_timed_out")
             is CancellationException -> message("codewhisperer.codescan.cancelled_by_user_exception")
-            else -> e.message
+            else -> when {
+                e.message?.startsWith("codewhisperer.codescan.run_scan_throttling_error_telemetry") == true -> message("codewhisperer.codescan.run_scan_throttling_error_telemetry")
```
Comment (ctlai95) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt:391]: Please add a comment about why this was added

**Review by ctlai95 — APPROVED** [review]

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt [hunk 5]
```diff
@@ -322,14 +322,26 @@ class CodeWhispererCodeScanManager(val project: Project) {
         return errorMessage
     }
 
+    private fun getCodeScanExceptionMessage(e: CodeWhispererCodeScanException): String? {
+        val message = e.message
+        return when {
+            message == message("codewhisperer.codescan.invalid_source_zip_telemetry") -> {
+                message("codewhisperer.codescan.run_scan_error")
+            }
+            message.isNullOrBlank() -> null
```
Comment (manodnyab) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt:331]: nit: Should this order be reversed, ie should isNullOrBlank be the the first condition?

File: plugins/toolkit/resources/resources/software/aws/toolkits/resources/MessagesBundle.properties [hunk 6]
```diff
@@ -761,6 +761,7 @@ codewhisperer.codescan.run_scan_complete=<html><body> Security Scan completed fo
 codewhisperer.codescan.run_scan_error=Amazon Q encountered an error while scanning for security issues. Please try again later.
 codewhisperer.codescan.run_scan_error_telemetry=Security scan failed.
 codewhisperer.codescan.run_scan_info=Select 'Run' in toolbar to scan this package for security issues.
+codewhisperer.codescan.run_scan_throttling_error_telemetry= Too many requests, please wait before trying again.
```
Comment (manodnyab) [inline plugins/toolkit/resources/resources/software/aws/toolkits/resources/MessagesBundle.properties:764]: Is this error user facing or is this error being sent to telemetry?

Comment (laileni-aws) [reply plugins/toolkit/resources/resources/software/aws/toolkits/resources/MessagesBundle.properties:764]: No this is sent to telemetry not customer facing message.

Comment (manodnyab) [reply plugins/toolkit/resources/resources/software/aws/toolkits/resources/MessagesBundle.properties:764]: This should probably not reside here then cause this file has user-facing text. Telemetry reason could be moved within the code to where we are sending the telemetry or a utils file.

Comment (laileni-aws) [reply plugins/toolkit/resources/resources/software/aws/toolkits/resources/MessagesBundle.properties]: Moved this text from `MessageBundle.properties` to the respective file.

**Pull request merged**